application/src/main/java/org/togetherjava/tjbot/commands/tags/TagManageCommand.java

```diff
@@ -280,6 +318,69 @@ private boolean isWrongTagStatusAndHandle(@NotNull TagStatus requiredTagStatus,
         return false;
     }
 
+    private void logAction(@NotNull Subcommand subcommand, @NotNull Guild guild,
+            @NotNull User author, @NotNull TemporalAccessor timestamp, @NotNull String id,
+            @Nullable String newContentNullable, @Nullable String previousContentNullable) {
+
+        Optional<String> newContent =
+                (newContentNullable == null) ? Optional.empty() : Optional.of(newContentNullable);
+        Optional<String> previousContent = (previousContentNullable == null) ? Optional.empty()
+                : Optional.of(previousContentNullable);
+
+        switch (subcommand) {
+            case CREATE -> ModAuditLogWriter.log("Tag-Manage Create",
+                    String.format("created tag **%s**", id), author, timestamp, guild,
+                    new ModAuditLogWriter.Attachment[] {new ModAuditLogWriter.Attachment(
+                            Filename.CONTENT.get(), newContent.orElseThrow())});
+
+            case CREATE_WITH_MESSAGE -> ModAuditLogWriter.log("Tag-Manage Create with message",
+                    String.format("created tag **%s**", id), author, timestamp, guild,
+                    new ModAuditLogWriter.Attachment[] {new ModAuditLogWriter.Attachment(
+                            Filename.CONTENT.get(), newContent.orElseThrow())});
+
+            case EDIT -> ModAuditLogWriter.log("Tag-Manage Edit",
+                    String.format("edited tag **%s**", id), author, timestamp, guild,
+                    new ModAuditLogWriter.Attachment[] {
+                            new ModAuditLogWriter.Attachment(Filename.NEW_CONTENT.get(),
+                                    newContent.orElseThrow()),
+                            new ModAuditLogWriter.Attachment(Filename.PREVIOUS_CONTENT.get(),
+                                    previousContent.orElseThrow())});
+
+            case EDIT_WITH_MESSAGE -> ModAuditLogWriter.log("Tag-Manage Edit with message",
+                    String.format("edited tag **%s**", id), author, timestamp, guild,
+                    new ModAuditLogWriter.Attachment[] {
+                            new ModAuditLogWriter.Attachment(Filename.NEW_CONTENT.get(),
+                                    newContent.orElseThrow()),
+                            new ModAuditLogWriter.Attachment(Filename.PREVIOUS_CONTENT.get(),
+                                    previousContent.orElseThrow())});
+
+            case DELETE -> ModAuditLogWriter.log("Tag-Manage Delete",
+                    String.format("delete tag **%s**", id), author, timestamp, guild,
+                    new ModAuditLogWriter.Attachment[] {new ModAuditLogWriter.Attachment(
+                            Filename.PREVIOUS_CONTENT.get(), previousContent.orElseThrow())});
```

Comment:
Never use `Optional#get`, instead always use `.orElseThrow()`. Both methods are 100% identical but former is deprecated due to bad naming.

Reply:
Seems that I got confused with the method naming.